### DISCUSSION TITLE
Is it possible somehow to change input's value on fly in uncontrolled mode?

### DISCUSSION THREAD

**Discussion by dmitry-ff** [mantinedev/mantine#7191] [Q&A] [answered]
I need to allow user enter only integers. By default input with `type={number}` allows user to enter `.`. I thought i can avoid it by checking user input value by regExp but its not help at all. As i understand it will works only in controlled forms.
Next things i tried - is to subscribe on field's changes and set prev value if input's value doesn't pass test. But now i lose field`s focus.

Maybe exists any solution how i can perform such behavior

**Comment by rtivital** [comment, marked as answer]
No, it is not possible with uncontrolled mode, you can either move this state manipulations down to component level or use controlled form mode.
